include/cutlass/predicate_vector.h

```diff
@@ -490,15 +490,20 @@ struct PredicateVector {
 
   /// Returns true if entire predicate array is zero.
   CUTLASS_HOST_DEVICE bool is_zero() const {
+    constexpr Storage byte_mask = (((1 << kPredicatesPerByte) - 1) << kPredicateStart);
     Storage mask(0);
     for (int byte = 0; byte < sizeof(Storage); ++byte) {
-      Storage byte_mask = (((1 << kPredicatesPerByte) - 1) << kPredicateStart);
       mask |= (byte_mask << (byte * 8));
     }
-    uint32_t result = 0;
-    for (int word = 0; word < kWordCount; ++word) {
-      result |= storage(word);
+    Storage result = 0;
+    for (int word = 0; word < kWordCount - 1; ++word) {
```

Comment:
why do we need to minus 1?

Reply:
To  exclude the last word which is handled below

Reply:
i see. thanks.

looking at the other methods in this struct, I think we can assume that we don't have a partial word.

Reply:
> i see. thanks.
> 
> looking at the other methods in this struct, I think we can assume that we don't have a partial word.

I disagree, may be there are hidden bugs in other methods because of word handling.

Reply:
the register size is always 32bit.  it does not make sense if predicatevector is not multiple of 32bit.  you can add an assert to assure it if you want.

Reply:
I mean kBytes may not be a multiple of 32bit. So we have to deal with the last word because it is possible to contain out-of-bound bytes.